src/crepe/system/InputSystem.cpp

```diff
@@ -0,0 +1,184 @@
+	for (Camera & cam : cameras) {
+		if (!cam.active) continue;
+		this->curr_cam_ref = cam;
+		break;
+	}
+	if (!curr_cam_ref) return;
+	Camera & current_cam = curr_cam_ref;
+	RefVector<Transform> transform_vec
+		= mgr.get_components_by_id<Transform>(current_cam.game_object_id);
+	Transform & cam_transform = transform_vec.front().get();
+	int camera_origin_x
+		= cam_transform.position.x + current_cam.offset.x - (current_cam.viewport_size.x / 2);
+	int camera_origin_y
+		= cam_transform.position.y + current_cam.offset.y - (current_cam.viewport_size.y / 2);
+
+	for (const SDLContext::EventData & event : event_list) {
+		int world_mouse_x = event.mouse_position.first + camera_origin_x;
+		int world_mouse_y = event.mouse_position.second + camera_origin_y;
+		// check if the mouse is within the viewport
+		bool mouse_in_viewport
+			= !(world_mouse_x < camera_origin_x
+				|| world_mouse_x > camera_origin_x + current_cam.viewport_size.x
+				|| world_mouse_y < camera_origin_y
+				|| world_mouse_y > camera_origin_y + current_cam.viewport_size.y);
+
+		switch (event.event_type) {
+			case SDLContext::EventType::KEYDOWN:
+				event_mgr.queue_event<KeyPressEvent>(KeyPressEvent{
+					.repeat = event.key_repeat,
+					.key = event.key,
+				});
+				break;
+			case SDLContext::EventType::KEYUP:
+				event_mgr.queue_event<KeyReleaseEvent>(KeyReleaseEvent{
+					.key = event.key,
+				});
+				break;
+			case SDLContext::EventType::MOUSEDOWN:
+				if (!mouse_in_viewport) {
+					break;
+				}
+				event_mgr.queue_event<MousePressEvent>(MousePressEvent{
+					.mouse_x = world_mouse_x,
+					.mouse_y = world_mouse_y,
+					.button = event.mouse_button,
+				});
+				last_mouse_down_position = {world_mouse_x, world_mouse_y};
+				last_mouse_button = event.mouse_button;
+				break;
+			case SDLContext::EventType::MOUSEUP: {
+				if (!mouse_in_viewport) {
+					break;
+				}
+				event_mgr.queue_event<MouseReleaseEvent>(MouseReleaseEvent{
+					.mouse_x = world_mouse_x,
+					.mouse_y = world_mouse_y,
+					.button = event.mouse_button,
+				});
+				//check if its a click by checking the last button down
+				int delta_x = world_mouse_x - last_mouse_down_position.first;
+				int delta_y = world_mouse_y - last_mouse_down_position.second;
+
+				if (last_mouse_button == event.mouse_button
+					&& std::abs(delta_x) <= click_tolerance
+					&& std::abs(delta_y) <= click_tolerance) {
+					event_mgr.queue_event<MouseClickEvent>(MouseClickEvent{
+						.mouse_x = world_mouse_x,
+						.mouse_y = world_mouse_y,
+						.button = event.mouse_button,
+					});
+
+					handle_click(event.mouse_button, world_mouse_x, world_mouse_y);
+				}
+			} break;
+			case SDLContext::EventType::MOUSEMOVE:
+				if (!mouse_in_viewport) {
+					break;
+				}
+				event_mgr.queue_event<MouseMoveEvent>(MouseMoveEvent{
+					.mouse_x = world_mouse_x,
+					.mouse_y = world_mouse_y,
+					.rel_x = event.rel_mouse_move.first,
+					.rel_y = event.rel_mouse_move.second,
```

Comment:
**Volgens mij** zijn deze twee relative waardes nog in pixels en moet dit nog even om worden gezet naar game units

Reply:
hier moet ik het met niels nog even over hebben is het goed dat ik voor nu hier een miro ticket voor maak zodat de rest van de functionaliteit gemerged kan worden (ik weet ook niet of we het gaan gebruiken dus ik kan het ook weghalen voor nu)

Reply:
Prima om een Miro ticket te maken hoor